Readme.md

```diff
@@ -1,76 +1,37 @@
-
-## Special requirements for today's hackathon:
-
-*Aesthetics*: There is no visual component to today's project. Your score will be determined by your module's beauty. i.e. by the method names, number of parameters, whether they are functions or classes, namespaces within your package, and how lightweight the imported module is, etc.
-
-The more methods you put in your package, the better.
-
-*Deployment*:
-
-Videos and screenshots will not be accepted. You must leave links to:
-
-1. Your NPM package
-2. A deployed project on Runkit or Codepen.
-
-
-Your deployment options for #2 above are:
-
-1. Runkit - You will have to sign up and create a new notebook. See the example [Runkit Notebooks](https://runkit.com/monarchwadia). 
-2. Codepen - You can import your project into Codepen using [unpkg](https://unpkg.com/)
-
-## Bonus points
-
-Bonus points will be awarded for:
-* Creating a typescript-compatible project
-* Creating unit tests
-* Using a precompiler (only if appropriate, i.e. if you are using Typescript or Babel, or if you are creating a React library)
-* Leaving detailed instructions for local development on `Readme.md`
-* Writing documentation (for your future self's benefit)
+Javascript library with methods to loop through, create and clone multidimensional arrays
 
-## Frameworks:
+## Install
 
-Any or none. You can create a personal library with just plain JavaScript. You can also create a personal library that contains common React/Vue/Angular components that you find yourself using often across projects.
+`npm install @definitelynotsam/matrix-utils`
 
-## Restrictions
+## How to use
 
-* Your project must use JavaScript.
-* You must follow deployment instructions above.
+```js
+const matrixUtils = require('@definitelynotsam/matrix-utils');
 
-There are no other restrictions.
+const testMatrix = [
+	[-1, -1],
+	[-1, 0],
+	[-1, 1],
+	[0, 1],
+	[1, 1],
+	[1, 0],
+	[1, -1],
+	[0, -1],
+];
 
-## Tips for success:
+console.log(matrixUtils.loopMatrix(testMatrix));
 
-1. The project is intentionally simple so you can focus on code quality.
-1. The requirements are intentionally minimalistic so you can get as creative as you'd like.
-1. You get a LOT of time for code review comments. So you can win even if you feel your project isn't great, simply on the strength of your code review comments!
-1. Keep in mind, 33% of your score is based on aesthetic appeal!
+const newMatrix = matrixUtils.createMatrix(5, 5, () => Math.random() >= 0.6);
+console.log(newMatrix);
 
-## Resources
+const newMatrixClone = matrixUtils.cloneMatrix(newMatrix);
 
-* [Creating and publishing unscoped public packages](https://docs.npmjs.com/creating-and-publishing-unscoped-public-packages)
-* [Video - Creating & publishing your own NPM modules](https://www.youtube.com/watch?v=rTsz09zRuTU)
-* [freeCodeCamp's Tutorial](https://www.freecodecamp.org/news/how-to-make-a-beautiful-tiny-npm-package-and-publish-it-2881d4307f78/)
+console.log(newMatrix === newMatrixClone);
```

Comment:
We are expecting `false` here, right? Would be useful to leave comments with expected output for each `console.log` statement.